urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/XMLExampleHandler.java

```diff
@@ -0,0 +1,390 @@
+		}*/
+	
+
+	
+	// The XML parser only works with POST data
+	if( postData == null )  
+	    throw new DataFormatException( "You MUST send your XML document via HTTP POST!" );
+	    
+	
+	FormData formData        = postData.readFormData();
+	if( formData.size() == 0 )
+	    throw new HeaderFormatException( "No params passed (but required)." );
+
+
+	StringBuffer buffer    = new StringBuffer();
+	StringBuffer xmlBuffer = null;   // A buffer to store the XML data
+	String xmlFileName     = null;
+	buffer.append( "This is a test.\n" ).
+	    append( "\n" ).
+	    append( "The " + getClass().getName() + " class expects an XML document with the form item name 'my_file' passed via HTTP POST.\n" ).
+	    append( "\n\n" ).
+	    append( "Statistics:\n" ).
+	    append( "-----------\n" );
+	buffer.append( " =================== There are " + formData.size() + " form data items. ===============\n" );
+
+	for( int i = 0; i < formData.size(); i++ ) {
+
+	    FormDataItem item = formData.get(i);
+	    buffer.append( " +++ Item["+i+"]=" + item + "\n" );
+	    StringBuffer tmpXMLBuffer = new StringBuffer();
+	    String tmpFileName = this.processFormDataItem( item, i, buffer, tmpXMLBuffer );
+	    
+	    if( tmpFileName != null && xmlFileName == null ) {
+		xmlBuffer   = tmpXMLBuffer;
+		xmlFileName = tmpFileName;
+	    }
+
+	}
+
+
+	buffer.append( "\n\n\n" );
+	if( xmlBuffer == null )
+	    buffer.append( "No XML file sent.\n" );
+	else {
+	    buffer.append( "Received XML data from file '" ).
+		append( xmlFileName ).
+		append( "'.\n" ).
+		append( "---BEGIN---------- XML data ------------------\n" ).
+		append( xmlBuffer ).append( "\n" ).
+		append( "---END------------ XML data ------------------\n\n\n" );
+
+	    this.parseXMLData( buffer, xmlFileName, xmlBuffer.toString() );
+	}
+
+
+
+
+	buffer.append( "-EOI-" );
+	
+
+       
+
+
+	Session<String,BasicType,ikrs.httpd.HTTPConnectionUserID> session = this.getHTTPHandler().getSessionManager().get( sessionID );
+	
+
+	String data = buffer.toString();
+	buffer.delete( 0, buffer.length() ); // Clear buffer
+
+	ByteArrayResource resource = new ByteArrayResource( this.getHTTPHandler(),
+							    this.getLogger(),
+							    data.getBytes(),
+							    false   // no need to use fair locks
+							    );
+
+	// I want this output to be displayed as plain text.
+	// One other possible way would be
+	//   MIMEType mimeType = new MIMEType( "text/plain" );
+	MIMEType mimeType = MIMEType.getByFileExtension( "txt" );
+	resource.getMetaData().setMIMEType( mimeType );
+	resource.getMetaData().setCharsetName( "UTF-8" );
+		
+	return resource;
+    }
+    //--- END -------------------------- FileHandler implementation ------------------------------
+
+
+    /**
+     * If the 'my_file' param is found in the passed item the method will
+     * print the file data into the xmlBuffer and return the file's name
+     * (otherwise it returns null).
+     **/
+    private String processFormDataItem( FormDataItem item,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 242-251 into a separate method.